### PR TITLE
Xeno Update

### PR DESCRIPTION
**Pull request by Robustin**
:cl: Robustin
add: Facehuggers can now destroy face-blocking helmets. Facehuggers will deal a heavy mix of acid/brute damage to a helmet before dying. 
add: Xeno drones can now store facehuggers.
balance: Xeno drones also have more HP and can throw facehuggers farther than other xenos.
balance: Xeno hunters no longer have sanic speed, and their leap has less distance and takes longer to cool down - but they now crawl into vents and recover from failed leaps significantly faster.
/:cl:

**I'm still working out the kinks of the surprisingly frustrating task of giving storage back to Drones (I need advice on how to do it)**. It's a modest update that should make class choice feel more nuanced, make drone a more viable role, and make facehuggers a factor even for helmeted crew. 

FYI: The helmet damage ranged from 2 hits for a standard EVA helmet to something like 6 for a nukeop hardsuit.

### CODE REVIEW

File: code/modules/mob/living/carbon/alien/humanoid/update_icons.dm [hunk 1]
```diff
@@ -91,4 +92,12 @@
...
\ No newline at end of file
+	apply_overlay(HANDS_LAYER)
+
+/mob/living/carbon/alien/humanoid/update_inv_pockets(update_icons = 1)
+	if(l_store)
+		l_store.screen_loc = ui_alien_storage_l
+	if(r_store)
+		r_store.screen_loc = ui_alien_storage_r
+	if(update_icons)
+		update_icons()
```
Comment (ShizCalev) [inline code/modules/mob/living/carbon/alien/humanoid/update_icons.dm]: Add a new line at the end of this file if you won't mind.

File: code/modules/mob/living/carbon/alien/update_icons.dm [hunk 2]
```diff
@@ -8,4 +8,4 @@
 	return
 
 /mob/living/carbon/alien/update_body_parts()//we don't use the bodyparts layer for aliens.
-	return
\ No newline at end of file
+	return
```
Comment (ShizCalev) [inline code/modules/mob/living/carbon/alien/update_icons.dm]: new line

File: code/modules/mob/living/ventcrawling.dm [hunk 3]
```diff
@@ -5,7 +5,7 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, typecacheof(list(
 
 //VENTCRAWLING
 
-/mob/living/proc/handle_ventcrawl(atom/A)
+/mob/living/proc/handle_ventcrawl(atom/A, var/speed = 25)
```
Comment (ShizCalev) [inline code/modules/mob/living/ventcrawling.dm]: remove var/

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Robustin** [comment]
I might also give resin nests some restorative properties, with space being uber lethal now but Xenos being pretty dependent on space access, its a frustrating trying to keep hosts alive until they burst.

File: code/modules/mob/living/carbon/alien/humanoid/inventory.dm [hunk 4]
```diff
@@ -3,3 +3,40 @@
...
+				return FALSE
+			if(!(I.slot_flags))
+				return FALSE
+			return TRUE
+	return FALSE //Unsupported slot
+
+/mob/living/carbon/alien/humanoid/attack_ui(slot_id)
+	var/obj/item/I = get_active_held_item()
+	if(!I)
+		return 0
```
Comment (ShizCalev) [inline code/modules/mob/living/carbon/alien/humanoid/inventory.dm]: FALSE

File: code/modules/mob/living/carbon/alien/humanoid/inventory.dm [hunk 5]
```diff
@@ -3,3 +3,40 @@
...
+	switch(slot_id)
+		if(slot_l_store)
+			if(l_store)
+				return 0
```
Comment (ShizCalev) [inline code/modules/mob/living/carbon/alien/humanoid/inventory.dm]: FALSE

File: code/modules/mob/living/carbon/alien/humanoid/inventory.dm [hunk 6]
```diff
@@ -3,3 +3,40 @@
...
+			l_store = I
+			update_inv_pockets()
+		if(slot_r_store)
+			if(r_store)
+				return 0
```
Comment (ShizCalev) [inline code/modules/mob/living/carbon/alien/humanoid/inventory.dm]: FALSE

File: code/modules/mob/living/carbon/alien/humanoid/inventory.dm [hunk 7]
```diff
@@ -3,3 +3,40 @@
...
+			r_store = I
+			update_inv_pockets()
+	I.equipped(src, slot_id)
+	return 1
```
Comment (ShizCalev) [inline code/modules/mob/living/carbon/alien/humanoid/inventory.dm]: TRUE

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by KorPhaeron** [comment]
>balance: Xeno hunters no longer have sanic speed

sad day

**Comment by pantherpusher** [comment]
So then what will protect against facehuggers?

**Comment by Spaceinaba** [comment]
still helmets but you'll have to replace them if you get too many huggers thrown at you

**Comment by WJohn** [comment]
I am strongly against making the throwing of facehuggers more prevalent. I think it's an outdated mechanic to make xenos viable and a better method could be developed.

Hunters losing their sanic speed is fine, but counterbalancing it with what is still unbelievably bullshit (their leaping) being made even more powerful than it already is is just ridiculous. Absolutely not happening.

Can't you already throw facehuggers like 7 tiles anyway? The bonus throw range means absolutely nothing.

Drones having a completely ridiculous 180 health means they can effectively face-tank just about anything thrown their way and get away with it due to their ability to wall off threats and huge plasma reserve. While this would make them more appealing, I don't think this is the right way to make them more of a dedicated support class (which is what I'd like to do with them). A dedicated ability to heal others (but not themselves) and view the health of friendly xenos would be a good way to handle this.

I'd like it if you checked out the idea pile I have for xenos here on how to perhaps refine them better.
https://pastebin.com/GJG6d1aL

**Comment by Buggy123** [comment]
TFW you make a PR that literally changes a 0 to a 1 and kick off a xeno revolution

:+1:

**Comment by kevinz000** [comment]
no, robustin. no.

**Comment by jughu** [comment]
Remember when sentinels were slow and auctally had to protect the hive instead of rush everything with their stunspit

**Comment by Shadowlight213** [comment]
Uh, don't you only have to keep a host alive until they are inpregnated? Not until they burst?
Or was that nerfed

**Comment by Robustin** [comment]
It used to be alive until burst, maybe it was changed, when I first touched xeno code I recall the impregnation was handled like a disease and diseases wouldn't process/progress on dead people.

**Comment by Robustin** [comment]
Also @kevinz000 

You're the reason I ignore emoji shit now. You shitposting in every feature PR, going as far as to rally people on Discord to also shitpost, just convinced me you're far more interested in being an asshole than improving the game. 

Grow the fuck up.

**Comment by Robustin** [comment]
5 is the throw range. It becomes 7 for drones. 

@KorPhaeron 

I know you're joking but sanic hunters was precisely why you closed #17180.

@WJohn 

Colonial Marines actually took power away from other Xeno abilities and pushed it toward facehugging. It worked great. It based CM not another iteration of stunspam combat but a siege-type game where you had to strategically deplete the other side's resources. Facehuggers should be a threat and their versatility in combat is wonderfully nuanced compared to 99% other our other combat mechanics. 

180hp is not "facetanking", its 2 more bloody lasers. I'd consider dropping it if Drones could heal but I'm not going forward with that until I have a better sense of where this PR is going. 

Leap is still leap, the stun is still long enough to do 120 damage when hunters have 125 max hp. Without sanic speed I didn't want leap to be a death sentence if you misclicked/lagged/whatever. **I find it ironic that you're shitting on leap so hard like it was my idea when I've been opposed to leap for years while your reaction to it being added was, and I quote, "I am very happy" "👍 💯".**

I would consider tweaking up the cooldown on leap but I am worried that if I make it too cumbersome to use then people just go sentinel and hunter becomes the new dead caste.

**Comment by epochayur** [comment]
>balance: Xeno drones also have more HP and can throw facehuggers farther than other xenos.
>balance: Xeno hunters no longer have sanic speed, but they now crawl into vents and recover from failed leaps significantly faster.

"balance"

**Comment by WJohn** [comment]
Resin doors promoted lazy design by just spamming them absolutely everywhere.

I would rather we make it so nests apply slowdown to anyone and anything moving over them, xenos included, so there is a reason to use them in chokes.

**Comment by Robustin** [comment]
I can't tell whether the screeching from Kev & Co. is because xenos are too weak or too strong to him but having weeds slow down would be a big issue inside of nests for the Xenos, unless it didn't apply to Xenos, in which case it would be a major buff to Xenos.

**Comment by WJohn** [comment]
I mean specifically resin NESTS, the thing you strap people down to. Not the weeds in general. This'd give them a dual purpose of both restraining people AND acting as a sort of doorway.

**Comment by PKPenguin321** [comment]
dislike the helmets change personally

**Comment by Armhulen** [comment]
Remove xenomorphs

**Comment by Robustin** [comment]
The whole points of experimenting with this PKP is to see if we can shift away from the current (silly) system of stun spam and stripping people and move toward facehuggers being a more core mechanic.

I have yet to see a full xeno takeover this year. Worst case is everyone gets out because a Xeno problem is getting worse and nobody wants to deal with it, but generally they just get stomped as soon as security finds out where the Queen is. 

Earlier this week we had a xeno, he evolved to Praetorian before I found him, I opened the door, emptied a laser into his slow unwieldy ass, then used the organs for free thermals and shit. It's gotten to the point that my primary reaction to Xenos is "YAY FREE ORGAN PARTY!" instead of "OH SHIT WELD THE VENTS".

**Comment by WJohn** [comment]
And tell me, how does this solve really bad players continuing to be bad? None of this touches upon praetorians here.

Tips, information, and perhaps even tutorials are the way to help bad players. Not only that, but having multiple xenos at the start means there's more of a chance for one to be decent.

**Comment by Robustin** [comment]
Because I was commenting on Xenos power as a whole.

I find it funny that I'm this many comments in and you're still the only one who has written a coherent response on the substance of this.

**Comment by Armhulen** [comment]
i thought "remove xenomorphs" was pretty coherent

**Comment by Fox-McCloud** [comment]
Xenos are already horrifically frustrating once they get a foothold--they have to be stopped quickly (or cheesed via something like health-sensor max cap bombs), or they just turn into this deathball that no one can really do anything about.

This is only going to worsen that, only the time gap where it's viable to fight them will be even smaller.

Bad xenos die quickly, but when it's someone who even *half* knows how to play them? There's not much that can be done *against* them.

Having a more counter-play heavy xenos would be a great step towards moving them away from "lulz round ending event" to something that could hypothetically be its own gamemode.

**Comment by DoctoreMad** [comment]
Oh great so now we have to call the shuttle because the last xeno invasion broke all hardsuit helmets?

**Comment by Buggy123** [comment]
The way I see it is, one part of the problem is xenos have a massive upper hand in attrition - a human out of the fight is, usually, 1 new xeno in the fight. Xenos alone or in small numbers are easily swarmed and killed despite their strengths; I've certainly seen my fare share of friendlyish xeno queens get slaughtered trying to leave on the shuttle, and that's a _queen_. But xenos replicate quickly and turn a opposing tide of bodies into a _advantage_, and usually the crew just doesn't have enough meat to fight off a infestation.

Also, I don't remember what I said before, but why was it deleted?

**Comment by uraniummeltdown** [comment]
Xenos ard already absurdly strong, the leap is just broken

**Comment by Robustin** [comment]
Xenos are absurdly strong largely because hunters regenerate HP, move at nigh-unhittable speeds, and can't be stunned.

I don't get the impression anyone feels that any other Xeno caste is "OP", and this is largely a Hunter nerf with a consolation prize thrown in. 

I could increase the leap cooldown length but this PR feels like its turned into a political shitfest where I still have no idea where most people stand on Xenos. Arguments like "OH SO NOW WE GOTTA CALL THE SHUTTLE ONCE WE LOSE EVERY FACE-PROTECTING HELMET ON THE STATION" just reeks of someone desperately looking for an excuse to not like something.

A basic tech disk can be exported in exchange for over 100 riot helmets. Autolathes can print welding masks too. Try to come up with a more creative shitpost.

**Comment by Robustin** [comment]
I nerfed leap some more since thats the only coherent feedback I've gotten so far.

**Comment by PKPenguin321** [comment]
>The whole points of experimenting with this PKP is to see if we can shift away from the current (silly) system of stun spam and stripping people

what makes this silly? makes sense to me

**Comment by Robustin** [comment]
Because if you've ever played CM you'd realize how much better the "helmet as your HP bar" system works. It promotes a style of Xeno v. Human combat that is slower and more methodical than "1 and done" stun combat that typically involves a hunter dragging someone around at warp speed tackle spamming until they can strip the helmet. 

Alternatively we will NEVER have decent Xeno combat while sanic hunters + stunhands exist. There is nothing that brings sanity to that. Yet if we just strip sanic speed then a couple sec with riot shields basically become Xeno crushing gods. 

It's so silly that there's still controversy about Xeno balance. Someone came along and balance handed the SS13 community a template on how to make Xeno combat FUN and draw in a large number of players. Yet even the slightest shift away from our failed blueprint of "lightspeed hunters with stunhands are the backbone of xeno combat" and you get REEEEEEEEEEEEEEEEEE'd louder than the deathgasp spam when an atmos tech drags a plasma can into a xeno hive.

**Comment by PKPenguin321** [comment]
i guess that's fair, but it's not like this prevents xenos from stunning and stripping anyways

**Comment by optimumtact** [comment]
wow people hate this

**Pull request closed without merging**